sdks/go/pkg/beam/io/filesystem/filesystem.go

```diff
@@ -92,6 +93,12 @@ type Interface interface {
 // The following interfaces are optional for the filesystems, but
 // to support
 
+// LastModifiedGetter is an interface for getting the last modified time
+// of a file.
+type LastModifiedGetter interface {
```

Comment:
Let me know if you have a preference for whether this method should be part of `filesystem.Interface` or a standalone interface. (And if the latter, if the name `LastModifiedGetter` is idiomatic enough or if you can think of any better?)

Reply:
When I last updated the design around this, the intent was to have several as optional methods as interfaces since some operations may be infeasible for a given files system. Adding to the core interface is a backwards incompatible change for (the granted, likely few) users who have a private implementation, even if they're all updated here.